applications/StructuralMechanicsApplication/custom_utilities/tangent_operator_calculator_utility.h

```diff
@@ -414,23 +462,25 @@ class TangentOperatorCalculatorUtility
         double& rMinValue
         )
     {
-        const std::size_t size1 = rMatrixValues.size1();
-        const std::size_t size2 = rMatrixValues.size2();
-        std::vector<double> non_zero_values;
-
-        for (std::size_t i = 0; i < size1; ++i) {
-            for (std::size_t j = 0; j < size2; ++j) {
-                if (std::abs(rMatrixValues(i, j)) > tolerance)
-                    non_zero_values.push_back(std::abs(rMatrixValues(i, j)));
+       // Sizes of the matrix
+        const SizeType size1 = rMatrixValues.size1();
+        const SizeType size2 = rMatrixValues.size2();
+
+        // The deformation gradient is an identity matrix for zero deformation
+        const Matrix working_matrix = rMatrixValues - IdentityMatrix(size1);
```

Comment:
now you are allocating a matrix... the point is not to allocate anything!!

Reply:
I do the operation F - I everytime?